tests/brevitas_examples/test_llm.py

```diff
@@ -0,0 +1,442 @@
+    params=[
+        ModelAndPpl(
+            name="hf-internal-testing/tiny-random-LlamaForCausalLM",
+            float_ppl=None,
+            supports_fx=True,
+        ),
+        ModelAndPpl(
+            name="hf-internal-testing/tiny-random-MistralForCausalLM",
+            float_ppl=None,
+            supports_fx=False,
+        ),
+        #ModelAndPpl(
+        #    name="dacorvo/Mixtral-tiny",
+        #    float_ppl=None,
+        #    supports_fx=True,
+        #),
+    ])
+def small_models_with_ppl(request):
+    yield request.param
+
+
+@pytest.fixture()
+def default_run_args(request):
+    args = UpdatableNamespace(**vars(parse_args([])))
+    args.nsamples = 2
+    args.seqlen = 2
+    args.model = "hf-internal-testing/tiny-random-MistralForCausalLM"
+    args.dataset = "c4"
+    args.eval = True
+    #args.checkpoint = ptid2pathname(request.node.nodeid) + ".pth" # Example filename which won't clash
+    args.export_prefix = ptid2pathname(request.node.nodeid)
+    args.weight_bit_width = 8
+    args.weight_quant_granularity = "per_channel"  # "per_tensor", "per_channel", "per_group".
+    args.input_bit_width = 8
+    args.act_calibration = True
+    return args
+
+
+@pytest.fixture(
+    params=[
+        {},
+        {
+            "bias_corr": True},
+        {
+            "act_equalization": "layerwise"},
+        {
+            "act_equalization": "fx"},
+        {
+            "weight_equalization": True},
+        {
+            "gptq": True},
+        {
+            "ln_affine_merge": True},])
+def toggle_run_args(default_run_args, request):
+    args = default_run_args
+    args.update(**request.param)
+    yield args
+
+
+@pytest.mark.llm
+@requires_pt_ge('2.2')
+def test_small_models_toggle_run_args(caplog, toggle_run_args, small_models_with_ppl):
+    caplog.set_level(logging.INFO)
+    args = toggle_run_args
+    args.model = small_models_with_ppl.name
+    exp_float_ppl = small_models_with_ppl.float_ppl
+    use_fx = requires_fx(args)
+    if use_fx and not small_models_with_ppl.supports_fx:
+        pytest.xfail(f"{small_models_with_ppl.name} does not support FX")
+    float_ppl, quant_ppl, model = main(args)
+
+
+@pytest.fixture(
+    scope="session",
+    params=[
+        ModelAndPpl(
+            name="hf-internal-testing/tiny-random-OPTForCausalLM",
+            float_ppl=None,
+            supports_fx=True,
+        ),])
+def small_models_with_ppl_pt_ge_2_4(request):
+    yield request.param
+
+
+@pytest.mark.llm
+@requires_pt_ge('2.4')
+def test_small_models_toggle_run_args_pt_ge_2_4(
+        caplog, toggle_run_args, small_models_with_ppl_pt_ge_2_4):
+    caplog.set_level(logging.INFO)
+    args = toggle_run_args
+    args.model = small_models_with_ppl_pt_ge_2_4.name
+    exp_float_ppl = small_models_with_ppl_pt_ge_2_4.float_ppl
+    use_fx = requires_fx(args)
+    if use_fx and not small_models_with_ppl_pt_ge_2_4.supports_fx:
+        pytest.xfail(f"{small_models_with_ppl.name} does not support FX")
```

Comment:
This will never xfail currently, correct? Can we add a test case to stress this or just remove it?

Reply:
I'll refactor this to share the same code with the `2.2` one. It makes more sense with that context.